encodings/runend/src/compute/mod.rs

```diff
@@ -116,15 +116,24 @@ impl TakeFn<RunEndArray> for RunEndEncoding {
 
 impl SliceFn<RunEndArray> for RunEndEncoding {
     fn slice(&self, array: &RunEndArray, start: usize, stop: usize) -> VortexResult<ArrayData> {
-        let slice_begin = array.find_physical_index(start)?;
-        let slice_end = array.find_physical_index(stop)?;
+        let new_length = stop - start;
+
+        let (slice_begin, slice_end) = if new_length == 0 {
+            let values_len = array.values().len();
+            (values_len, values_len)
```

Comment:
I'm not 100% sure that all array types let you slice starting at the exclusive end... might get complaints about out of bounds?

Reply:
But I think they should? It's valid right.

Reply:
I think that all arrays should have the same slicing semantics behavior, such as:
```rust
slice(array, array.len(), array.len().unwrap().is_empty()
```
is always true/doesn't panic.